plotly/tools.py

```diff
@@ -1443,6 +1449,333 @@ class FigureFactory(object):
     """
 
     @staticmethod
+    def _unlabel_rgb(colors):
+        unlabelled_colors = []
+        for color in colors:
+            str_vals = ''
+            for index in range(len(color)):
+                try:
+                    float(color[index])
+                    str_vals = str_vals + color[index]
+                except ValueError:
+                    if (color[index] == ',') or (color[index] == '.'):
+                        str_vals = str_vals + color[index]
+
+            str_vals = str_vals + ','
+            numbers = []
+            str_num = ''
+            for char in str_vals:
+                if char != ',':
+                    str_num = str_num + char
+                else:
+                    numbers.append(float(str_num))
+                    str_num = ''
+            unlabelled_tuple = (numbers[0], numbers[1], numbers[2])
+            unlabelled_colors.append(unlabelled_tuple)
+
+        return unlabelled_colors
+
+    @staticmethod
+    def _find_intermediate_color(tuple1, tuple2, t):
+        # Given two color tuples (in normalized RGB space)
+        # and a value 0 < t < 1, spit out a color that is
+        # t percent from tuple1 to tuple2.
+        diff_0 = float(tuple2[0] - tuple1[0])
+        diff_1 = float(tuple2[1] - tuple1[1])
+        diff_2 = float(tuple2[2] - tuple1[2])
+
+        new_tuple = (tuple1[0] + t*diff_0,
+                     tuple1[1] + t*diff_1,
+                     tuple1[2] + t*diff_2)
+        return new_tuple
+
+    @staticmethod
+    def _unconvert_from_RGB_255(colors):
+        un_rgb_colors = []
+        for color in colors:
+            un_rgb_color = (color[0]/(255.0),
+                            color[1]/(255.0),
+                            color[2]/(255.0))
+
+            un_rgb_colors.append(un_rgb_color)
+        return un_rgb_colors
+
+    @staticmethod
+    def _map_z2color(zval, colormap, vmin, vmax):
+        # Map the normalized value zval to a
+        #corresponding color in the colormap
+        if vmin > vmax:
+            raise ValueError("Incorrect relation between vmin and vmax."
+                             " The vmin cannot be bigger than vmax.")
+        t = (zval - vmin)/float((vmax - vmin))  # normalize val
+        t_color = FigureFactory._find_intermediate_color(colormap[0],
+                                                         colormap[1],
+                                                         t)
+        t_color = (t_color[0]*255.0, t_color[1]*255.0, t_color[2]*255.0)
+        labelled_color = 'rgb{}'.format(t_color)
+
+        return labelled_color
+
+    @staticmethod
+    def _tri_indices(simplices):
+        return ([triplet[c] for triplet in simplices] for c in range(3))
+
+    @staticmethod
+    def _plotly_trisurf(x, y, z, simplices, colormap=None,
```

Comment:
🐄 `_get_trisurf_data`?

I don't think we need to keep `_plotly_*`-ing the names of these. It's clear that the user is using plotly to generate this stuff, IMO.